crates/system-manager/src/main.rs

```diff
@@ -37,10 +100,9 @@ struct Args {
     /// The host to deploy the system-manager profile to
     target_host: Option<String>,
 
-    #[arg(long, action)]
-    /// Invoke the remote command with sudo.
-    /// Only useful in combination with --target-host
-    use_remote_sudo: bool,
+    #[arg(long = "use-remote-sudo", action, hide = true)]
+    /// Deprecated: use --sudo on the subcommand instead
+    legacy_use_remote_sudo: bool,
```

Comment:
There is a `#[deprecated]` attribute where you can include a reason, might be useful here.